src/main/java/org/zalando/nakadi/service/EventStream.java

```diff
@@ -47,7 +47,7 @@ public EventStream(final EventConsumer eventConsumer,
         this.writer = writer;
     }
 
-    public void streamEvents(final AtomicBoolean connectionReady) {
+    public void streamEvents(final AtomicBoolean connectionReady, final Runnable checkAuthorization) {
```

Comment:
@v-stepanov Initially I did it like that. but because of the difference of how it calculated (and called on zk thread, actually), I have to perform authorization check here. It will require adding several additional parameters here (eventTypeCache + authorizationProvider), which, in turn will require modification of tons of unit tests, that are all passing all the parameters to construct object. It is too much, that is why I decided to go with runnable (I do not like it as well)